app/src/main/java/com/pvp/app/ui/common/Components.kt

```diff
@@ -208,4 +220,204 @@ fun TextField(
         },
         messages = errors
     )
+}
+
+@Composable
+fun ButtonWithDialog(
+    modifier: Modifier = Modifier,
+    contentAlignment: Alignment = Alignment.TopStart,
+    mainButtonContent: @Composable RowScope.() -> Unit = { Text("Open Dialog") },
+    confirmButtonContent: @Composable RowScope.() -> Unit = { Text("Confirm") },
+    dismissButtonContent: @Composable RowScope.() -> Unit = { Text("Dismiss") },
+    dialogTitle: @Composable () -> Unit = { Text("Dialog Title") },
+    dialogContent: @Composable () -> Unit = { Text("Dialog Content") },
+    onConfirmClick: () -> Unit = {},
+    onDismiss: () -> Unit = {}
+) {
+    var showDialog by remember { mutableStateOf(false) }
+
+    Box(
+        modifier = modifier,
+        contentAlignment = contentAlignment
+    ) {
+        androidx.compose.material3.Button(
```

Comment:
Could use our own button (which is above) to keep being consistent with some of the modifications in the button.

For now our button pretty much changes the shape and possibly something else (I do not recall), in the future we may change it even more. To keep writing those modifications in every spot is a non-sense, so that's why we should create a modified component and re-use it.